integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/search/loading/SearchQueryEntityLoadingBaseIT.java

```diff
@@ -62,6 +63,22 @@ public void simple() {
 		);
 	}
 
+	@Test
+	public void simple_withTimeout() {
+		final int entityCount = 3;
+
+		persistThatManyEntities( entityCount );
+
+		testLoadingThatManyEntities(
+				session -> { }, // No particular session setup
+				o -> { }, // No particular loading option
+				entityCount,
+				// Only one entity type means only one statement should be executed, even if there are multiple hits
+				c -> c.assertStatementExecutionCount().isEqualTo( 1 ),
+				1, TimeUnit.SECONDS
```

Comment:
You can't just put a timeout and not make any assertion related to that timeout. The test, as it is, is just a smoke test.

You need to test at least these cases:

* `failAfter` with a very large timeout: expect everything to work fine.
* `failAfter` with a small-ish timeout (say, 1ms) and very slow loading: expect an exception to be thrown.
* `failAfter` with a large timeout (say, 500ms), very slow index response (say, 499ms) and normal loading: expect an exception to be thrown.
* And maybe more. Have a look at your code and see if some obvious edge cases are not covered by the above?

To create the right conditions for your tests (slow index response time, slow loading), you have two options.

The cleanest approach would be to stub the `TimingSource`. However, it might not be very easy, and if done incorrectly you may end up not testing what you want.

The admittedly more dirty, but more straightforward approach is to voluntarily slow down your components:

* To simulate slow index response time, just pass a custom behavior to `org.hibernate.search.util.impl.integrationtest.common.rule.BackendMock#expectSearchObjects(java.util.Collection<java.lang.String>, java.util.function.Consumer<org.hibernate.search.util.impl.integrationtest.common.stub.backend.search.StubSearchWork.Builder>, org.hibernate.search.util.impl.integrationtest.common.rule.StubSearchWorkBehavior<org.hibernate.search.engine.backend.common.DocumentReference>)`, and in that behavior call `Thread.sleep()`.
* To simulate slow loading, you can add a `SessionEventListener` through `SessionBuilderImpl#eventListeners`, and in `SessionEventListener#jdbcPrepareStatementEnd()` call `Thread.sleep()`.

Reply:
I added extensive (I think) for the first and second cases. About the last case you mentioned. IMO this would test the time capability of the stub backend to track time and to pass the right difference to the mapper. I don't thing it would make much sense.